File: src/src_platform_jfx/fmg/jfx/img/Mine.java

package fmg.jfx.img;

import fmg.core.img.LogoModel;

/** Mine image on the playing field */
public final class Mine {

    /** Mine image controller implementation for {@link Logo.Canvas} */
    public static class ControllerCanvas extends Logo.ControllerCanvas {
        public ControllerCanvas() { LogoModel.toMineModel(getModel()); }
    }

    /** Mine image controller implementation for {@link Logo.Image} */
    public static class ControllerImage extends Logo.ControllerImage {
        public ControllerImage() { LogoModel.toMineModel(getModel()); }
    }

}

File: FastMines_uwp/src/fmg/uwp/img/win2d/Mine.cs
using Microsoft.Graphics.Canvas;
using Fmg.Core.Img;

namespace Fmg.Uwp.Img.Win2d {

    /// <summary> Mine image on the playing field. Win2D implementation </summary>
    public static class Mine {

        /// <summary> Mine image controller implementation for <see cref="Logo.CanvasBmpView"/> </summary>
        public class CanvasBmpController : Logo.CanvasBmpController {
            public CanvasBmpController(ICanvasResourceCreator resourceCreator)
                : base(resourceCreator)
            {
                LogoModel.ToMineModel(Model);
            }
        }

        /// <summary> Mine image controller implementation for <see cref="Logo.CanvasImgSrcView"/> </summary>
        public class CanvasImgSrcController : Logo.CanvasImgSrcController {
            public CanvasImgSrcController(ICanvasResourceCreator resourceCreator)
                : base(resourceCreator)
            {
                LogoModel.ToMineModel(Model);
            }
        }

    }

}
